### PR TITLE
Update kubernetes.rst

### LINKED ISSUE #13006: Setting in_cluster to True is failing with invalid kube config
Airflow version: 1.10.12

Deploying the example https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/example_dags/example_spark_kubernetes.py and

Added parameter  **in_cluster=True** in the dag. 

Expected Result:
Dag gets executed and spark application is deployed in the local cluster where airflow is deployed

Actual Result:
Error: {taskinstance.py:1150} ERROR - Invalid kube-config file. No configuration found.

Work-around:
Works after manually adding a kube config in the worker pod.

### PR DESCRIPTION
**Pull request by ecerulm**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow-providers-cncf-kubernetes/connections/kubernetes.rst [hunk 1]
```diff
@@ -20,7 +20,8 @@
 Kubernetes cluster Connection
 =============================
 
-The Kubernetes cluster Connection type enables connection to Kubernetes cluster.
+The Kubernetes cluster Connection type enables connection to Kubernetes cluster by SparkOperator only.
+This has nothing to do with KubernetesPodOperator.
```
Comment (jedcunningham) [inline docs/apache-airflow-providers-cncf-kubernetes/connections/kubernetes.rst]:
```suggestion
The Kubernetes cluster Connection type enables connection to a Kubernetes cluster by ``SparkKubernetesOperator`` tasks. They are not used by ``KubernetesPodOperator`` tasks.
```

**Review by jedcunningham — APPROVED** [review]
Looks good @ecerulm. Can you try rebasing on master to get the test green?

**Comment by kaxil** [comment]
Should be fixed after https://github.com/apache/airflow/pull/15076 is merged and the PR is rebased on master

File: docs/apache-airflow-providers-cncf-kubernetes/connections/kubernetes.rst [hunk 2]
```diff
@@ -20,7 +20,7 @@
 Kubernetes cluster Connection
 =============================
 
-The Kubernetes cluster Connection type enables connection to Kubernetes cluster.
+The Kubernetes cluster Connection type enables connection to a Kubernetes cluster by ``SparkKubernetesOperator`` tasks. They are not used by ``KubernetesPodOperator`` tasks.
```
Comment (mik-laj) [inline docs/apache-airflow-providers-cncf-kubernetes/connections/kubernetes.rst]:
```suggestion
The Kubernetes cluster Connection type enables connection to a Kubernetes cluster by :class:`~airflow.providers.cncf.kubernetes.operators.spark_kubernetes.SparkKubernetesOperator` tasks. They are not used by ``KubernetesPodOperator`` tasks.
```

**Review by kaxil — APPROVED** [review]

**Pull request merged**